### PR TITLE
[BUGFIX] Make BeforeSearchResultIsShownEvent more usable

### PR DESCRIPTION
**Pull request by georgringer**
It would be cool if properties of the event could be changed. Otherwise the event doesn't help much - IMO

I need it e.g. for the pagination to be changed

### CODE REVIEW

File: Classes/Event/Search/BeforeSearchResultIsShownEvent.php [hunk 1]
```diff
@@ -79,4 +80,9 @@ public function getCurrentPage(): int
     {
         return $this->currentPage;
     }
+
+    protected function setPagination(PaginationInterface $pagination): void
```
Comment (dkd-kaehm) [inline Classes/Event/Search/BeforeSearchResultIsShownEvent.php:84]: Why not public?

File: Classes/Event/Search/BeforeSearchResultIsShownEvent.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 use ApacheSolrForTypo3\Solr\Domain\Search\ResultSet\SearchResultSet;
 use ApacheSolrForTypo3\Solr\Pagination\ResultsPagination;
```
Comment (dkd-kaehm) [inline Classes/Event/Search/BeforeSearchResultIsShownEvent.php:21]: This line should is not needed anymore.

**Review by dkd-kaehm — CHANGES_REQUESTED** [review]
Thanks for this patch.
There are most probably two changes required.

**Review by dkd-kaehm — APPROVED** [review]
LGTM!
The PR can be squashed and merged.

**Comment by georgringer** [comment]
@dkd-kaehm I don't know if other properties should be setable as well but certainly all other events must be changed as well to be usable for extension authors. I suggest to use a different PR for that

**Comment by georgringer** [comment]
@dkd-kaehm wanna merge that?

**Comment by bmack** [comment]
Can we add them to the other Events that are related to Actions/Controllers?

**Pull request merged**